src/finch/tensor.py

```diff
@@ -139,13 +164,23 @@ def preprocess_order(
 
         return permutation
 
-    def get_order(self, zero_indexing=True) -> tuple[int, ...]:
+    @classmethod
+    def get_lvl_ndim(cls, lvl: JuliaObj) -> int:
+        ndim = 0
+        for _ in range(np.MAXDIMS):
```

Comment:
32 dims is a limitation by NumPy, we shouldn't limit ourselves to that necessarily.

```suggestion
        while True:
```

Reply:
Done!